File: Problemset/0844-Backspace String Compare/Stack.py
class Solution:
    def backspaceCompare(self, S: str, T: str) -> bool:
        stack_s = []
        stack_t = []

        for ch in S:
            if ch != '#':
                stack_s.append(ch)
            elif stack_s:
                stack_s.pop()
        for ch in T:
            if ch != '#':
                stack_t.append(ch)
            elif stack_t:
                stack_t.pop()

        return stack_s == stack_t

File: Problemset/1209-Remove All Adjacent Duplicates in String II/Stack.rs
impl Solution {
    pub fn remove_duplicates(s: String, k: i32) -> String {
        let mut stack: Vec<(u8, i32)> = vec![];

        for ch in s.bytes() {
            match stack.last_mut() {
                Some(last) if ch == last.0 => {
                    last.1 += 1;
                    if last.1 == k {
                        stack.pop();
                    }
                }
                _ => stack.push((ch, 1)),
            }
        }

        String::from_utf8(
            stack
                .into_iter()
                .map(|(ch, cnt)| vec![ch; cnt as usize])
                .flatten()
                .collect(),
        )
        .unwrap()
    }
}
